Multisig/Features/Security/SecurityCenter.swift

```diff
@@ -182,19 +182,26 @@ class SecurityCenter {
         let toggleWillBeOff: Bool = enabledOptions.contains(passcodeOption)
         if toggleWillBeOff {
             requestPasswordV2(for: [protectionClass]) { [unowned self] plaintextPasscode in
+                let currentDerivedPassword = plaintextPasscode.map { derivedKey(from: $0) }
+                try protectedKeyStore.authenticate(password: currentDerivedPassword)
```

Comment:
I thought that authentication here is not needed, because the other call will fail

Reply:
Actually I added it because I thought change password doesn't check for correct passcode, then I discovered that it's not working properly, I will remove it